FidelityInternational/endpoint-service-operator
# Endpoint service operator POC

## What

endpoint service operator POC

## How to test

- deploy EKS + cf for k8s
- clone the repo and check out branch
- export AWS_REGION
- execute `make run`

### What you should see

- operator will receive an initial batch of events from k8s API.
- it will filter them and pick only events from k8s services that are LoadBalancer type
- it will try to provision VPC Endpoint Service, VPC endpoint and DNS records for all filtered services
- it will say NO NO as it needs kube service annotations that configure VPC ID for endpooint and hosted zoone ID for DNS records

## How to provide annotations - 2 options

- create new service - paste this and modify
```
apiVersion: v1
kind: Service
metadata:
 annotations:
   service.beta.kubernetes.io/aws-load-balancer-internal: "true"
   service.beta.kubernetes.io/aws-load-balancer-type: nlb
   vpc-endpoint-service-operator.fil.com/endpoint-vpc-id: [cloudops provided VPC ID]
   vpc-endpoint-service-operator.fil.com/hosted-zone-id: [hostedZoneID  from route53]
   vpc-endpoint-service-operator.fil.com/hostname: [test.cf.dev66.whatevathezoneTLDis]
 name: test
 namespace: default
spec:
 ports:
 - name: https
   port: 443
   protocol: TCP
   targetPort: 8443
 type: LoadBalancer
 ```
now do kubectl appy and watch operator logs

- if you want to test on existing service you can do:
```
kubectl -n istio-system edit svc istio-ingressgateway
```
end add these lines to the annotations section
```
    vpc-endpoint-service-operator.fil.com/endpoint-vpc-id: [cloudops provided VPC ID]
    vpc-endpoint-service-operator.fil.com/hosted-zone-id: [hostedZoneID  from route53]
    vpc-endpoint-service-operator.fil.com/hostname: [api.cf.dev66.whatevathezoneTLDis]
```

## Test deletion 
```
kubectl delete -f service.yaml
```
service yaml is the service that you deployed in earlier steps. This should delete DNS records, VPC endpoint and VPC endpoint service. LBs are deleted by kube API

## Who
Not @combor or @alibowl or @StuWill
